libbeat/docs/shared-ssl-config.asciidoc

```diff
@@ -211,157 +340,225 @@ key: |
+==== `verification_mode`
 
-The following elliptic curve types are available:
+Controls the verification of client certificates. Valid values are:
 
-* P-256
-* P-384
-* P-521
-* X25519
-
-[float]
-==== `renegotiation`
+`full`::
+Verifies that the provided certificate is signed by a trusted
+authority (CA) and also verifies that the server's hostname (or IP address)
+matches the names identified within the certificate.
 
-This configures what types of TLS renegotiation are supported. The valid options
-are `never`, `once`, and `freely`. The default value is never.
+`strict`::
+Verifies that the provided certificate is signed by a trusted
+authority (CA) and also verifies that the server's hostname (or IP address)
+matches the names identified within the certificate. If the Subject Alternative
+Name is empty, it returns an error.
 
-* `never` - Disables renegotiation.
-* `once` - Allows a remote server to request renegotiation once per connection.
-* `freely` - Allows a remote server to repeatedly request renegotiation.
+`certificate`::
+Verifies that the provided certificate is signed by a
+trusted authority (CA), but does not perform any hostname verification.
 
+`none`::
+Performs _no verification_ of the server's certificate. This
+mode disables many of the security benefits of SSL/TLS and should only be used
+after very careful consideration. It is primarily intended as a temporary
+diagnostic mechanism when attempting to resolve TLS errors; its use in
+production environments is strongly discouraged.
++
+The default value is `full`.
 
 [float]
-==== `ca_sha256`
+[[server-renegotiation]]
+==== `renegotiation`
 
-This configures a certificate pin that you can use to ensure that a specific certificate is part of the verified chain.
+This configures what types of TLS renegotiation are supported. The valid options
+are:
 
-The pin is a base64 encoded string of the SHA-256 of the certificate.
+`never`::
+Disables renegotiation.
 
-NOTE: This check is not a replacement for the normal SSL validation, but it adds additional validation.
-If this option is used with  `verification_mode` set to `none`, the check will always fail because
-it will not receive any verified chains.
+`once`::
+Allows a remote server to request renegotiation once per connection.
 
+`freely`::
+Allows a remote server to repeatedly request renegotiation.
++
+The default value is `never`.
 
 ifeval::["{beatname_lc}" == "filebeat"]
 [float]
+[[server-client-renegotiation]]
 ==== `client_authentication`
 
-This configures what types of client authentication are supported. The valid options
-are `none`, `optional`, and `required`. When `certificate_authorities` is set it will
-default to `required` otherwise it will be set to `none`.
+The type of client authentication mode. When `certificate_authorities` is set, it
+defaults to `required`, otherwise it defaults to `none`.
+
+[NOTE]
+=====
+This option is only valid with the TCP or the Syslog input.
+=====
```

Comment:
Not sure how true this is. We also have one http server based input in filebeat and metricbeat each.

Reply:
From the docs, it looks like it's also valid for HTTP Endpoint, HTTP JSON, and MQTT.

Reply:
These are Filebeat specific. Do we use the same doc for auditbeat as is? Maybe we should just remove the note?

Reply:
I agree, let's remove the note.